paddle/fluid/framework/details/all_reduce_deps_pass.cc

```diff
@@ -48,19 +48,71 @@ std::unique_ptr<ir::Graph> AllReduceDepsPass::ApplyImpl(
   auto graph_ops = ir::FilterByNodeWrapper<OpHandleBase>(*graph);
 
   // get vars order
-  int order = 0;
   std::unordered_map<std::string, int> vars;
-  // TODO(gongwb): use graph topology sort to find the order of operators.
-  //               Note that must assert topology sort is stable
   auto& ops = Get<const std::vector<OpDesc*>>(kAllOpDescs);
-  for (auto* op_desc : ops) {
-    auto outputs = op_desc->Outputs();
-    for (auto& o_it : outputs) {
-      for (auto& v : o_it.second) {  // values
-        vars[v] = order;
+
+  // TODO(liangdun): add a config in build strategy
+  bool use_topological_sort = true;
```

Comment:
This pass is disable by default: https://github.com/PaddlePaddle/Paddle/blob/67093da398c6e06e0656c1fc1e9909cf3846df1d/paddle/fluid/framework/details/build_strategy.cc#L33-L35, only need sorting ops in python?

Reply:
It can be used with parallel_graph, what do you mean "sorting ops in python"?